src/evict/evict_lru.c

```diff
@@ -1355,7 +1353,7 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 		 * no good eviction candidates can be found.  Abandon the walk
 		 * if we get into that situation.
 		 */
-		give_up = pages_seen > 100 &&
+		give_up = !__wt_cache_aggressive(session) && pages_seen > 100 &&
```

Comment:
I did see that - it seems reasonable to me. I don't think we need to split it out.